src/linalg/impl_linalg.rs

```diff
@@ -532,6 +595,32 @@ fn blas_compat_1d<A, S>(a: &ArrayBase<S, Ix1>) -> bool
 }
 
 #[cfg(feature="blas")]
+fn blas_compat<A, S, D>(a: &ArrayBase<S, D>) -> bool
+    where S: Data,
+          A: 'static,
+          S::Elem: 'static,
+          D: Dimension,
+{
+    if !same_type::<A, S::Elem>() {
+        return false;
+    }
+
+    match D::equispaced_stride(&a.raw_dim(), &a.strides) {
```

Comment:
Ok, so the way the `equispaced_stride` method is implemented right now is that it grabs the smallest stride and calculates the rest based on that. That means that if we indeed have one stride set to 0, the stride information will always be 0 irrespective of the dimension (because its multiplicative). It follows that the demanded stride will never match the rest of the stride info in the struct. Thus the function will return `None` and we won't be able to use blas.